packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -180,33 +195,46 @@ class RenderParagraph extends RenderBox {
     // us to validate our intrinsic sizes, which will change _textPainter's
     // layout because the intrinsic size calculations are destructive.
     final Size textSize = _textPainter.size;
-    size = constraints.constrain(textSize);
+    final double constrainedHeight =
+      _maxLines == null ? textSize.height :
+        textSize.height.clamp(0.0, _textPainter.preferredLineHeight * _maxLines);
```

Comment:
Shouldn't the text painter have taken this into account already?  If you give the text painter a max lines, can it give you back a size that's bigger than that?